session/txnmanager.go

```diff
@@ -153,12 +174,29 @@ func (m *txnManager) EnterNewTxn(ctx context.Context, r *sessiontxn.EnterNewTxnR
 	if r.Type == sessiontxn.EnterNewTxnWithBeginStmt {
 		m.sctx.GetSessionVars().SetInTxn(true)
 	}
+
+	m.recordEvent("enter txn")
 	return nil
 }
 
 func (m *txnManager) OnTxnEnd() {
 	m.ctxProvider = nil
 	m.stmtNode = nil
+
+	m.events = append(m.events, event{event: "txn end", duration: time.Since(m.lastInstant)})
+
+	duration := time.Since(m.enterTxnInstant)
+	threshold := m.sctx.GetSessionVars().SlowTxnThreshold
+	if threshold > 0 && uint64(duration.Milliseconds()) >= threshold {
+		logutil.BgLogger().Info(
```

Comment:
Print connection id if possible is better as `BgLogger` is used.

Reply:
Connection id has been included